taichi/transforms/auto_diff.cpp

```diff
@@ -45,7 +45,7 @@ class IndependentBlocksJudger : public BasicStmtVisitor {
     is_inside_loop_ = false;
   }
 
-  static bool run(IRNode *root) {
+  static void run(IRNode *root, bool &is_ib, bool &is_smallest_ib) {
```

Comment:
aha is it possible to make a IBType enum to manager this? ;)

Reply:
I think there is an inclusion relation between `ib` and `smallest ib`, i.e., a `smallest ib` must be an `ib`. Will a IBType enum help with this?